internal/controller/accessrequest_controller.go

```diff
@@ -156,6 +162,46 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return result, nil
 }
 
+type AccessRequestConflictError struct {
+	message string
+}
+
+func (e *AccessRequestConflictError) Error() string {
+	return e.message
+}
+
+func NewAccessRequestConflictError(msg string) *AccessRequestConflictError {
+	return &AccessRequestConflictError{
+		message: msg,
+	}
+}
+
+// Validate will verify if there are existing AccessRequests for the same
+// user/app/role already in progress.
+func (r *AccessRequestReconciler) Validate(ctx context.Context, ar *api.AccessRequest) error {
+	arList, err := r.findAccessRequestsByUserApp(ctx,
+		ar.GetNamespace(),
+		ar.Spec.Subject.Username,
+		ar.Spec.Application.Name,
+		ar.Spec.Application.Namespace)
+	if err != nil {
+		return fmt.Errorf("error finding AccessRequests by user and app: %w", err)
+	}
+	for _, arResp := range arList.Items {
+		// skip if it is the same AccessRequest or if the role is different
+		if (arResp.GetName() == ar.GetName() &&
+			arResp.GetNamespace() == ar.GetNamespace()) ||
+			arResp.Spec.RoleTemplateName != ar.Spec.RoleTemplateName {
+			continue
+		}
+		if arResp.Status.RequestState == api.GrantedStatus ||
+			arResp.Status.RequestState == api.RequestedStatus {
```

Comment:
Do we need to handle the case where `RequestState` is empty (maybe not reconciled yet?) or does it default to one of these two values?

Reply:
Fair point! Added a condition validating when `RequestState` is empty. Good catch!